app/services/actions/foreman/puppet_class/import.rb

```diff
@@ -0,0 +1,47 @@
+module Actions
+  module Foreman
+    module PuppetClass
+      class Import < Actions::EntryAction
+        def resource_locks
+          :import_puppetclasses
+        end
+
+        def run
+          # #obsolete_and_new can return nil if there's no change so we have to be careful with to_sentence
+          output[:errors] = ::PuppetClassImporter.new.obsolete_and_new(input[:changed]).try(:to_sentence)
+        end
+
+        def humanized_input
+          return nil if input[:changed].nil?
+
+          humanized_input = []
+          humanized_input << _('Add') + ' ' + format_env_and_classes_input(input[:changed][:new]) if input[:changed][:new].present?
```

Comment:
Right, I also moved this to be humanized_output, since this input could be quite long. I fixed classes count to be correctly parsed and to be printed only when it make sense, e.g. when env is destroy we only get information that it was destroy with all of its classes so we don't print the classes number (would be always 1)